### ISSUE TITLE
Remove websql and SystemJS support

### ISSUE THREAD

**Issue by pleerock** [typeorm/typeorm#1918]
@daniel-lang I would like to open a discussion if we really need to support a websql. websql was a feature to support typeorm in browser however it does have multiple limitations:

* only chrome-like browsers support it which means it can't be used in a real production scenarios
* lot of serious issues with it (like problems in schema sync, inconsistent behaviour, etc.)

Then we got a sqljs implementation which does not have any of above issues. So, shall we deprecate a websql and make sqljs as our first-citizen browser driver. Same I guess will work for cordova, ionic, etc., right?

What is your thoughts here?

Also btw I wanted to discuss a deprecation of systemjs support (since almost nobody use it) and make webpack-first support.

**Comment by pleerock** [comment]
btw I just checked sqljs implementation and have a question. Why in

https://github.com/typeorm/typeorm/blob/6d0507233b78342db0acecbd39a5f48955af7cbe/src/driver/sqljs/SqljsDriver.ts#L85

we have a fileExist check? Shouldn't it create a database if it does not exist? I mean what if I just created app without having a db yet?

Because I wasn't be able to setup a basic app:

```js
const typeorm = require("typeorm");
const Post = require("./post.js").Post;

typeorm.createConnection({
    type: "sqljs",
    autoSave: true,
    synchronize: true,
    location: __dirname  + "/db.sqlite",
    entities: [
        Post
    ]
})
    .then(() => console.log("done"))
    .catch(err => console.error(err));
```

**Comment by daniel-lang** [comment]
The only reason to keep websql has been that is nice to have when developing an ionic app in the browser. But sqljs would work just as fine (or even better considering the limitations) in this situation.
As mention in [this comment (#1081)](https://github.com/typeorm/typeorm/issues/1081#issuecomment-346166472), the current state of the driver is unusable for anything beyond a simple test.

SystemJS is a similar story. I recently tested `0.2` and noticed, that, like websql, it's only usable for simple tests because we have `require`-statements in multiple places to prevent circular dependencies. But those `require`-statements cannot be interpreted by SystemJS. So this is another indicator, that it is not used. Another thing that I'm quiet unhappy with in regards to SystemJS is that the file that tsc creates only works up to a certain version of SystemJS (newer version are not supported).

TL:DR; I'm all for dropping Websql and SystemJS support

**Comment by pleerock** [comment]
> TL:DR; I'm all for dropping Websql and SystemJS support

then probably I'll do that. But we need to pack umd module I think instead of systemjs module we have.

As you can see [0.2.0 roadmap](https://github.com/typeorm/typeorm/issues?q=is%3Aopen+is%3Aissue+milestone%3A0.2.0) is almost finished and we are planning to finish everything tomorrow with @AlexMesser . Most of tasks yours 😆 Please close which are resolved, which you think you can do today or tomorrow (deadline) please resolve them, which needs a discussion please let me know, others which you think are not possible in these two days please move to 0.3.0

**Comment by daniel-lang** [comment]
Regarding your comment about Sql.js:
Sql.js is an in-memory database which you can export and save into a file. Sql.js itself has no mechanism to have to this file.
This `fileExist()`-check is there to see if the file already exists from previous runs and load it if it exists, as Sql.js can load a database into memory.

I hope, that clears that up.

**Comment by daniel-lang** [comment]
> But we need to pack umd module I think instead of systemjs module we have.

I will check how easy it would be to integrate UMD.

> then probably I'll do that. 

Do you want to remove it or give a deprecation warning and remove it with 0.3?

**Comment by pleerock** [comment]
hard remove 🙅‍♂️  don't lot of people are using websql or systemjs. Not sure about people who are using typeorm on mobiles, its your area 😆

**Comment by daniel-lang** [comment]
hard remove is probably the option. Do you want to do it or should I create a PR for it?

**Comment by pleerock** [comment]
I was thinking to it by myself and then ask you to check everything, but if you think if it would be a better idea to do it by yourself, then go ahead

**Comment by pleerock** [comment]
> This fileExist()-check is there to see if the file already exists from previous runs and load it if it exists, as Sql.js can load a database into memory.

but it throw an exception and Im not able to run app. With following configuration (see above) I expected that will load database from the file if it exist, and if not it will simply write to the file once I save something. But im not able to run my app

**Comment by daniel-lang** [comment]
I don't care. Either way is fine with me.

**Comment by pleerock** [comment]
> /Users/pleerock/www/demos/react-app/desktop/node_modules/sql.js/js/sql.js:3 writeStringToMemory is deprecated and should not be called! Use stringToUTF8() instead!

btw I have such warning when using sqljs

**Comment by pleerock** [comment]
```
node src/app.js 
Error: File /Users/pleerock/www/demos/react-app/desktop/src/db.sqlite does not exist
    at SqljsDriver.load (/Users/pleerock/www/demos/react-app/desktop/node_modules/typeorm/driver/sqljs/SqljsDriver.js:135:27)
    at SqljsDriver.createDatabaseConnection (/Users/pleerock/www/demos/react-app/desktop/node_modules/typeorm/driver/sqljs/SqljsDriver.js:254:25)
    at SqljsDriver.<anonymous> (/Users/pleerock/www/demos/react-app/desktop/node_modules/typeorm/driver/sqljs/SqljsDriver.js:83:51)
    at step (/Users/pleerock/www/demos/react-app/desktop/node_modules/typeorm/driver/sqljs/SqljsDriver.js:42:23)
    at Object.next (/Users/pleerock/www/demos/react-app/desktop/node_modules/typeorm/driver/sqljs/SqljsDriver.js:23:53)
    at /Users/pleerock/www/demos/react-app/desktop/node_modules/typeorm/driver/sqljs/SqljsDriver.js:17:71
    at new Promise (<anonymous>)
    at __awaiter (/Users/pleerock/www/demos/react-app/desktop/node_modules/typeorm/driver/sqljs/SqljsDriver.js:13:12)
    at SqljsDriver.connect (/Users/pleerock/www/demos/react-app/desktop/node_modules/typeorm/driver/sqljs/SqljsDriver.js:77:16)
    at Connection.<anonymous> (/Users/pleerock/www/demos/react-app/desktop/node_modules/typeorm/connection/Connection.js:146:58)
```

code is above

**Comment by daniel-lang** [comment]
You right, it will throw an exception because of this

https://github.com/typeorm/typeorm/blob/6d0507233b78342db0acecbd39a5f48955af7cbe/src/driver/sqljs/SqljsDriver.ts#L90

I guess the idea when I wrote was, that if you use `SqljsEntityManager.load()` with a file that doesn't exist,  an exception should be thrown. This idea is still correct but as the same method is used to load the file on startup, it throws an exception. I will create a PR for it.
